sdks/python/apache_beam/examples/wordcount.py

```diff
@@ -111,6 +113,10 @@ def format_result(word_count):
 
   output = counts | 'format' >> beam.Map(format_result)
 
+  out_dir = os.path.dirname(known_args.output)
+  if not FileSystems.exists(out_dir):
```

Comment:
I believe the expectation should be that any output path should be created during pipeline execution and not by the driver program creating the pipeline.

Please revert this change to wordcount and fix the filesystem implementation to create any necessary directories instead.

Reply:
interesting, I originally [made a change to `LocalFileSystem` to create directories on `open`](https://github.com/ryan-williams/beam/commit/25868025c2ead0b695d0dde46b6d4e3d19a4923a), but I wasn't sure if that was the right semantics; it sounds like you're saying it is?

Reply:
I filed https://github.com/apache/beam/pull/5903 with that change; can close this in favor of that if that's what you prefer, thanks!

Reply:
Thanks, taking a look at #5903.